### ISSUE TITLE
option in Ansible provisioner to setup rsa, dsa or ed25519 passphrase key

### ISSUE THREAD

**Issue by ysolis** [lxdock/lxdock#99]
I am using now LXDock and we say in my work "bye Vagrant" :)

One thing that I note is that by default with Ansible provisioner using ssh (default) the rsa public key is exported to the container.

Will be posible to setup this and change to another keys used ? I use mostly ed25519 when i need to login to remote servers, in some cases, old CentOS 6 servers for example, I can only use rsa, and I use in this case rsa-4096 keys.

In general can be a good option for future deployments (for example, different keys for dev/test/prod.. containers)

**Comment by hsoft** [comment]
It would probably be even better for lxdock to generate its own private key like Vagrant does. But an even better option would be to use ansible's lxd transport option (#94). If my pull request at https://github.com/ansible/ansible/pull/26922 can manage to ever be merged, I don't think there would be any reason to use SSH with lxdock at all.

**Comment by robvdl** [comment]
@hsoft that would be really nice, one thing I was wondering as well, right now is there anything that can be done to get Ansible not to use root by default to login.  This is pretty strange to me and means I can't currently share my Ansible scripts with lxdock.yml and Vagrantfile because lxdock runs Ansible as root so everything is "inverted" in my Ansible scripts, it starts of as root and "becomes" the app user while it should be the other way around, login as the "app" user and become root for only some operations.

**Comment by hsoft** [comment]
@robvdl That would be an interesting feature to add, yes. And it's funny that you mention this because it would not work under the LXD transport: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/lxd.py#L46

So I think you've just found a reason why we would possibly want to keep the SSH transport around!

**Comment by robvdl** [comment]
Yes so what happens when you do lxdock up is it only copies the SSH key over for the root user right now so the root user can get in passwordless.  It would be necessary to do this for all users that are created via the lxdock.yml file for this to work, or something like that.
